docs/quick_start/installation.rst

```diff
@@ -536,3 +536,48 @@ Note: For other user facing environment variables, see :ref:`using_env_vars`.
     for Windows, Linux and macOS respectively. For systems that supports it,
     it is also possible to edit the RPATH of the plugin to add the location of
     the shared OpenColorIO lib.
+
+Updating python type stubs
+==========================
+
+The python stubs are built by analyzing the docstrings generated by pybind11
+and are used to drive IDE code completion and static analysis.  Changes to
+the bindings that result in changes to the stubs should be reviewed together
+in a single pull request, so that regressions are not unwittingly introduced.
+
+The workflow looks like this:
+
+- Install [`uv`](https://docs.astral.sh/uv/getting-started/installation/) and
+  `docker`
+- Create the temp build directory, cd into it, and run `cmake /path/to/source`
+- Run `cmake --build . --target pystubs` to generate updated stubs in
+  `src/bindings/python/stubs/__init__.pyi`
+- Commit the new stubs and push to Github
+- In CI, the stubs will be included in the wheels built by `cibuildwheel`, as
+  defined in `.github/workflows/wheel_workflow.yml`
+- In CI, one of the `cibuildwheel` Github actions will rebuild the stubs to a
+  temp location and verify that they match what has been committed to the repo.
+  This step ensures that if changes to the C++ source code and bindings results
+  in a change to the stubs, developers are notified of the need to regenerate
+  the stubs, so that changes can be reviewed and the rules in `generate_stubs.py`
+  can be updated, if necessary.
+
+Note that if you can't (or don't want to) build the stubs locally, you can
+download an artifact containing the wheel and `__init__.pyi` file from any job
+that fails the stub validation.
+
+If you want to bypass the use of docker, you can use the following steps to
+build the python bindings and run the stub generator directly:
+
+```
+python -m venv .venv
+. .venv/bin/activate
+mkdir build
+cd build
+cmake -DPython_EXECUTABLE=$(which python) -DCMAKE_INSTALL_PREFIX=../dist ..
+cmake --build . --target=install
+# extract the deps from the pyproject.toml and install them
+pip install $(yq -r '.tool.cibuildwheel.overrides.[0].test-requires' -oy ../pyproject.toml)
+PY_VER=$(python -c "import sys;print(f'{sys.version_info[0]}.{sys.version_info[1]}')")
+PYTHONPATH=../dist/lib/python$PY_VER/site-packages python ../src/bindings/python/stubs/generate_stubs.py --out-path ../src/bindings/python/stubs/
+```
```

Comment:
@zachlewis   I added instructions on how to build the stubs without using docker